warn/warn.go

```diff
@@ -142,39 +147,86 @@ var FileWarningMap = map[string]func(f *build.File) []*LinterFinding{
 	"unused-variable":           unusedVariableWarning,
 }
 
-// LegacyFileWarningMap lists the warnings that run on the whole file with legacy interface.
-var LegacyFileWarningMap = map[string]func(f *build.File, fix bool) []*Finding{}
-
 // nonDefaultWarnings contains warnings that are enabled by default because they're not applicable
 // for all files and cause too much diff noise when applied.
 var nonDefaultWarnings = map[string]bool{
 	"out-of-order-load":   true, // load statements should be sorted by their labels
 	"unsorted-dict-items": true, // dict items should be sorted
 }
 
-// RuleWarning is a wrapper that converts a per-rule function to a per-file function. It also doesn't
-// run on .bzl of default files.
-func RuleWarning(ruleWarning func(call *build.CallExpr) []*LinterFinding) func(f *build.File) []*LinterFinding {
-	return func(f *build.File) []*LinterFinding {
+// fileWarningWrapper is a wrapper that converts a file warning function to a generic function.
+// A generic function takes a `pkg string` argument which is not used for file warnings, so it's just removed.
+func fileWarningWrapper(fct func(f *build.File) []*LinterFinding) func(f *build.File, pkg string) []*LinterFinding {
+	return func(f *build.File, pkg string) []*LinterFinding {
+		return fct(f)
+	}
+}
+
+// ruleWarningWrapper is a wrapper that converts a per-rule function to a per-file function.
+// It also doesn't run on .bzl of default files.
```

Comment:
of default?

Reply:
there are 4 file types defined: BUILD, WORKSPACE, .bzl, and "default" (for just Starlark files, no Bazel-specific warnings should be raised for them).

Reply:
I see the typo now, will fix.